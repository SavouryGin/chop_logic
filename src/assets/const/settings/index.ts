import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
};
