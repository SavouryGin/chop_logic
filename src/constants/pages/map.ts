import React from 'react';
import { Pages } from 'constants/pages/ids';
import { paths } from 'constants/pages/paths';

type RoutesMapItem = {
  id: Pages;
  url: string;
  element: React.FC;
  title?: string;
};

const Predicates = React.lazy(() => import('pages/predicates'));
const TruthTables = React.lazy(() => import('pages/truth-tables'));
const Propositions = React.lazy(() => import('pages/propositions'));
const Syllogisms = React.lazy(() => import('pages/syllogisms'));

const routesMap: RoutesMapItem[] = [
  {
    id: Pages.Propositions,
    url: paths[Pages.Propositions],
    element: Propositions,
    title: 'Propositions',
  },
  {
    id: Pages.Predicates,
    url: paths[Pages.Predicates],
    element: Predicates,
    title: 'Predicates',
  },
  {
    id: Pages.TruthTables,
    url: paths[Pages.TruthTables],
    element: TruthTables,
    title: 'Truth Tables',
  },
  {
    id: Pages.Syllogisms,
    url: paths[Pages.Syllogisms],
    element: Syllogisms,
    title: 'Syllogisms',
  },
];

export default routesMap;
