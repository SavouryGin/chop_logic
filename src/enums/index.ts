export enum Icon {
  Default = 'chop-icon__check',
  Home = 'chop-icon__home',
  Menu = 'chop-icon__menu',
  Enlarge = 'chop-icon__enlarge2',
  Shrink = 'chop-icon__shrink2',
  Settings = 'chop-icon__cog',
  Delete = 'chop-icon__bin2',
  Up = 'chop-icon__circle-up',
  Down = 'chop-icon__circle-down',
  Right = 'chop-icon__circle-right',
  Left = 'chop-icon__circle-left',
  Sound = 'chop-icon__volume-medium',
  NoSound = 'chop-icon__volume-mute2',
  LightMode = 'chop-icon__sun',
  DarkMode = 'chop-icon__contrast',
}

export enum Pages {
  Home,
  NotFound,
  Propositions,
  Predicates,
  TruthTables,
  Syllogisms,
}
