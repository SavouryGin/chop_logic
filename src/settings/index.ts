import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const languageStringOptions = ['en', 'ru'];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  linkedIn: 'https://www.linkedin.com/in/dmitrii-suroviagin/',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
  facebook: 'https://www.facebook.com/dmitrii.suroviagin',
};
