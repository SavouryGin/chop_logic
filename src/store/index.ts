import createSagaMiddleware from 'redux-saga';
import rootReducer from './root-reducer';
import rootSaga from './root-saga';
import { configureStore } from '@reduxjs/toolkit';

const sagaMiddleware = createSagaMiddleware();

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat([sagaMiddleware]),
});

sagaMiddleware.run(rootSaga);
