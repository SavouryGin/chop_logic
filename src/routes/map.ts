import Home from 'basic-pages/home';
import Predicates from 'basic-pages/predicates';
import Propositions from 'basic-pages/propositions';
import TruthTables from 'basic-pages/truth-tables';
import Syllogisms from 'basic-pages/syllogisms';
import { Pages } from 'constants/pages/ids';
import { paths } from 'constants/pages/paths';

type RoutesMapItem = {
  id: Pages;
  path: string;
  element: React.FC;
};

const routesMap: RoutesMapItem[] = [
  {
    id: Pages.Home,
    path: paths[Pages.Home],
    element: Home,
  },
  {
    id: Pages.Propositions,
    path: paths[Pages.Propositions],
    element: Propositions,
  },
  {
    id: Pages.Predicates,
    path: paths[Pages.Predicates],
    element: Predicates,
  },
  {
    id: Pages.TruthTables,
    path: paths[Pages.TruthTables],
    element: TruthTables,
  },
  {
    id: Pages.Syllogisms,
    path: paths[Pages.Syllogisms],
    element: Syllogisms,
  },
];

export default routesMap;
