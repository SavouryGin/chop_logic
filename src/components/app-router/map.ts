import React from 'react';
import { Icon, Page } from 'enums';
import { paths } from './paths';

type RoutesMapItem = {
  id: Page;
  url: string;
  element: React.FC;
  title: string;
  key: string;
  icon?: Icon;
};

const Predicates = React.lazy(() => import('pages/predicates'));
const TruthTables = React.lazy(() => import('pages/truth-tables'));
const Propositions = React.lazy(() => import('pages/propositions'));
const Syllogisms = React.lazy(() => import('pages/syllogisms'));

export const routesMap: RoutesMapItem[] = [
  {
    id: Page.Propositions,
    key: `page-${Page.Propositions}`,
    url: paths[Page.Propositions],
    element: Propositions,
    title: 'Propositions',
    icon: Icon.Propositions,
  },
  {
    id: Page.Predicates,
    key: `page-${Page.Predicates}`,
    url: paths[Page.Predicates],
    element: Predicates,
    title: 'Predicates',
    icon: Icon.Predicates,
  },
  {
    id: Page.TruthTables,
    key: `page-${Page.TruthTables}`,
    url: paths[Page.TruthTables],
    element: TruthTables,
    title: 'Truth Tables',
    icon: Icon.TruthTables,
  },
  {
    id: Page.Syllogisms,
    key: `page-${Page.Syllogisms}`,
    url: paths[Page.Syllogisms],
    element: Syllogisms,
    title: 'Syllogisms',
    icon: Icon.Syllogisms,
  },
];
