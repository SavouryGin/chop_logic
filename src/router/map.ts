import React from 'react';
import { Icon, Page } from 'enums';
import { RoutesMapItem } from 'types';
import { paths } from './paths';

const Predicates = React.lazy(() => import('pages/predicates'));
const TruthTables = React.lazy(() => import('pages/truth-tables'));
const PropositionsIntroduction = React.lazy(() => import('pages/propositions/sub-pages/introduction'));
const PropositionsDirect = React.lazy(() => import('pages/propositions/sub-pages/direct-proofs'));
const PropositionsNatural = React.lazy(() => import('pages/propositions/sub-pages/natural-proofs'));
const Syllogisms = React.lazy(() => import('pages/syllogisms'));

export const routesMap: RoutesMapItem[] = [
  {
    id: Page.PropositionsIntro,
    key: `page-${Page.PropositionsIntro}`,
    url: paths[Page.PropositionsIntro],
    element: PropositionsIntroduction,
    title: { en: 'Propositions', ru: 'Высказывания' },
    icon: Icon.Propositions,
  },
  {
    id: Page.PropositionsDirectProofs,
    key: `page-${Page.PropositionsDirectProofs}`,
    url: paths[Page.PropositionsDirectProofs],
    element: PropositionsDirect,
    title: { en: 'Direct Proofs', ru: 'Прямые доказательства' },
    icon: Icon.File,
  },
  {
    id: Page.PropositionsNaturalProofs,
    key: `page-${Page.PropositionsNaturalProofs}`,
    url: paths[Page.PropositionsNaturalProofs],
    element: PropositionsNatural,
    title: { en: 'Natural Proofs', ru: 'Структурные доказательства' },
    icon: Icon.Files,
  },
  {
    id: Page.Predicates,
    key: `page-${Page.Predicates}`,
    url: paths[Page.Predicates],
    element: Predicates,
    title: { en: 'Predicates', ru: 'Предикаты' },
    icon: Icon.Predicates,
  },
  {
    id: Page.TruthTables,
    key: `page-${Page.TruthTables}`,
    url: paths[Page.TruthTables],
    element: TruthTables,
    title: { en: 'Truth Tables', ru: 'Таблицы' },
    icon: Icon.TruthTables,
  },
  {
    id: Page.Syllogisms,
    key: `page-${Page.Syllogisms}`,
    url: paths[Page.Syllogisms],
    element: Syllogisms,
    title: { en: 'Syllogisms', ru: 'Силлогизмы' },
    icon: Icon.Syllogisms,
  },
];
