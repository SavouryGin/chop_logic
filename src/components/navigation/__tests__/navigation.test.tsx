import Navigation from '../index';
import React from 'react';
import renderWithRedux from 'utils/test-utils/render-with-redux';
import { Page } from 'enums';
import { combineReducers } from '@reduxjs/toolkit';
import { routesMap } from 'router/map';
import { screen } from '@testing-library/react';
import { settingsInitialState, settingsSlice } from 'store/settings';

describe('Navigation component:', () => {
  const testProps = {
    className: 'test-class-name',
    isOpened: true,
  };

  const mockedReducer = combineReducers({ settings: settingsSlice.reducer });
  const mockedState = { settings: settingsInitialState };

  beforeEach(() => {
    renderWithRedux(<Navigation {...testProps} />, mockedReducer, mockedState);
  });

  it('renders a nav element', () => {
    const nav = screen.getByRole('navigation');
    expect(nav).toBeInTheDocument();
    expect(nav).toHaveClass('navigation', testProps.className);
  });

  it('contains the header', () => {
    const header = screen.getByText('Navigation', { exact: true });
    expect(header).toBeInTheDocument();
    expect(header).toHaveClass('navigation__header');
  });

  it('contains a list element', () => {
    const list = screen.getByRole('list');
    expect(list).toBeInTheDocument();
    expect(list).toHaveClass('navigation__list');
  });

  it('the amount of list items equals the number of top pages', () => {
    const pagesCount = routesMap.filter((item) => item.parentPageId === Page.Home).length;
    expect(screen.getAllByRole('listitem')).toHaveLength(pagesCount);
  });

  it('the list contains buttons to open nested links', async () => {
    const pagesCount = routesMap.length;
    const links = await screen.findAllByRole('link');
    expect(links).toHaveLength(pagesCount);
  });

  it('each list item displays the page title', async () => {
    const listItems = await screen.findAllByRole('listitem');
    for (let i = 0; i < listItems.length; i++) {
      expect(listItems[i]).toHaveTextContent(routesMap[i].title.en);
    }
  });

  it('each link has the page icon', async () => {
    const links = await screen.findAllByRole('link');
    for (let i = 0; i < links.length; i++) {
      expect(links[i]).toHaveClass(routesMap[i].icon);
    }
  });

  it('should match the snapshot', () => {
    const { asFragment } = renderWithRedux(<Navigation {...testProps} />, mockedReducer, mockedState);
    expect(asFragment()).toMatchSnapshot();
  });
});
