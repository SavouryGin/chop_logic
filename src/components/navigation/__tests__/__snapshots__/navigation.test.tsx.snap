// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component: should match the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navigation test-class-name"
  >
    <h2
      class="navigation__header"
    >
      Navigation
    </h2>
    <ul
      class="navigation__list"
    >
      <li>
        <span
          class="app-link"
          data-testid="link-to_2"
        >
          <a
            class="chop-icon__cube"
            href="/propositions"
          >
            Propositions
          </a>
        </span>
        <button
          class="show-more-button chop-icon__caret-down"
          data-testid="show-more-for_2"
          id="show-more-for_2"
          type="button"
        />
      </li>
      <li>
        <span
          class="app-link"
          data-testid="link-to_5"
        >
          <a
            class="chop-icon__cubes"
            href="/predicates"
          >
            Predicates
          </a>
        </span>
      </li>
      <li>
        <span
          class="app-link"
          data-testid="link-to_7"
        >
          <a
            class="chop-icon__dice"
            href="/syllogisms"
          >
            Syllogisms
          </a>
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
